wasswa-derick/Jambo
# #164172914 add dependency injection with dagger2

### What does this PR do?
- Create the retrofit instance for the network calls.

### Description of Task to be completed?

- [x]  add a dependency injection for the API service.
- [x]  add a base application that creates the modules ready for injection once the application is created.
- [x]  set the application in the manifest file
- [x]  add DI dependencies in the file

### How should this be manually tested?
``` Run the application and ensure that the modules are available for injection in the MainActivity```

### What are the relevant pivotal tracker stories?
[#164172914](https://www.pivotaltracker.com/story/show/164172914)

### Screenshots (if appropriate)

```N/A```
